libmcount/mcount.c

```diff
@@ -1182,6 +1182,9 @@ enum filter_result mcount_entry_filter_check(struct mcount_thread_data *mtdp, un
 	if (mcount_check_rstack(mtdp))
 		return FILTER_RSTACK;
 
+	if (mcount_min_size > 0 && mcount_getsize(&mcount_sym_info, child) < mcount_min_size)
+		return FILTER_OUT;
+
```

Comment:
This is only used when DISABLE_MCOUNT_FILTER is defined which means it's used without other filter/trigger options.  But I see you added the trigger action in the later patch.